### PR TITLE
fix: speed up ydb insert

### PR DESCRIPTION
**Pull request by ischasny**
- speed up ydb indexing by executing inserts in batches instead of concurrently
- add insert batch size database setting that is default to 10k

### Performance results

Car file used: `{"piececid": "bafykbzacedrrhwwbp7bcbvekiqzlzvbi65wy2ku6xkejlewz6bciasnnqycd4", "recs": 2232145}`

- The current implementation
```2023-11-01T13:06:20.289Z	WARN	boostd-data	cmd/run.go:339	added index to yugabytedb successfully	{"took": 244.186086292}```
- Batch inserts 1k batch size
```2023-11-01T12:51:23.415Z	WARN	boostd-data	cmd/run.go:348	added index to yugabytedb successfully	{"took": 47.338054792}```
- Batch inserts 10k batch size
```2023-11-01T12:53:26.419Z	WARN	boostd-data	cmd/run.go:348	added index to yugabytedb successfully	{"took": 40.236502708}```
- Batch inserts 50k batch size
```2023-11-01T12:52:23.827Z	WARN	boostd-data	cmd/run.go:348	added index to yugabytedb successfully	{"took": 35.621975375}```

### CODE REVIEW

File: extern/boostd-data/yugabyte/service.go [hunk 1]
```diff
@@ -73,6 +73,9 @@ func NewStore(settings DBSettings, migrator *Migrator, opts ...StoreOpt) *Store
 		settings: settings,
 		cluster:  cluster,
 		migrator: migrator,
+		// The Cassandra driver has a 50k limit on batch statements. Keeping
+		// batch size small makes sure we're under the limit.
+		cassandraBatchSize: 5000,
```
[the remark this answers is not in the document]
Comment (ischasny) [reply extern/boostd-data/yugabyte/service.go:78]: Agreed - will extract that into a separate setting if the perf optimisation proves to be legit

File: extern/boostd-data/yugabyte/service.go [hunk 2]
```diff
@@ -442,19 +445,36 @@ func (s *Store) addMultihashesToPieces(ctx context.Context, pieceCid cid.Cid, re
 	ctx, span := tracing.Tracer.Start(ctx, "store.add_index.payloadpiece")
 	defer span.End()
 
-	var count float64
-	return s.execParallel(ctx, recs, s.settings.PayloadPiecesParallelism, func(rec model.Record) error {
+	batchEntries := make([]gocql.BatchEntry, 0, len(recs))
```
[the remark this answers is not in the document]
Comment (ischasny) [reply extern/boostd-data/yugabyte/service.go:448]: Agreed. Will fix for both batch inserts

**Review by nonsense — APPROVED** [review]

File: extern/boostd-data/yugabyte/service.go [hunk 3]
```diff
@@ -442,54 +450,63 @@ func (s *Store) addMultihashesToPieces(ctx context.Context, pieceCid cid.Cid, re
...
-		if err != nil {
-			return fmt.Errorf("inserting into PayloadToPieces: %w", err)
+	insertPieceOffsetsQry := `INSERT INTO PayloadToPieces (PayloadMultihash, PieceCid) VALUES (?, ?)`
+	pieceCidBytes := pieceCid.Bytes()
+
+	var batch *gocql.Batch
+	for allIdx, rec := range recs {
+		if batch == nil {
+			batch = s.session.NewBatch(gocql.UnloggedBatch).WithContext(ctx)
+			batch.Entries = make([]gocql.BatchEntry, 0, s.settings.InsertBatchSize)
```
Comment (masih) [inline extern/boostd-data/yugabyte/service.go:460]: Size should be whatever is left, which is _at most_ `insertBatchSize`, not always.

File: extern/boostd-data/yugabyte/service.go [hunk 4]
```diff
@@ -442,54 +450,63 @@ func (s *Store) addMultihashesToPieces(ctx context.Context, pieceCid cid.Cid, re
...
 		}
 
-		count++
-		progress(count / float64(len(recs)))
-		return nil
-	})
+		batch.Entries = append(batch.Entries, gocql.BatchEntry{
+			Stmt:       insertPieceOffsetsQry,
+			Args:       []interface{}{trimMultihash(rec.Cid.Hash()), pieceCidBytes},
```
Comment (masih) [inline extern/boostd-data/yugabyte/service.go:465]:
```suggestion
			Args:       []any{trimMultihash(rec.Cid.Hash()), pieceCidBytes},
```

File: extern/boostd-data/yugabyte/service.go [hunk 5]
```diff
@@ -442,54 +450,63 @@ func (s *Store) addMultihashesToPieces(ctx context.Context, pieceCid cid.Cid, re
...
 
-	// The Cassandra driver has a 50k limit on batch statements. Keeping
-	// batch size small makes sure we're under the limit.
-	const batchSize = 5000
 	var batch *gocql.Batch
-	for allIdx, entry := range batchEntries {
+	for allIdx, rec := range recs {
 		if batch == nil {
 			batch = s.session.NewBatch(gocql.UnloggedBatch).WithContext(ctx)
+			batch.Entries = make([]gocql.BatchEntry, 0, s.settings.InsertBatchSize)
```
Comment (masih) [inline extern/boostd-data/yugabyte/service.go:493]: ditto re slice size.

Would be nice to reduce duplicate code in working with batches.

**Review by masih — APPROVED** [review]

**Pull request merged**